scim-sdk-server/src/main/java/de/captaingoldfish/scim/sdk/server/patch/PatchResourceHandler.java

```diff
@@ -89,24 +89,57 @@ public boolean addResourceValues(ObjectNode resource, JsonNode readJsonDocument,
       else
       {
         SchemaAttribute schemaAttribute = getSchemaAttribute((extensionUri == null ? "" : extensionUri + ":") + key);
-        verifyImmutableAndReadOnly(resource, schemaAttribute);
+        boolean resourceChanged;
+        boolean isReadOnly = isReadOnly(resource, schemaAttribute);
         if (Type.COMPLEX.equals(schemaAttribute.getType()))
         {
-          changeWasMade.weakCompareAndSet(false, addComplexAttribute(resource, schemaAttribute, value, extensionUri));
+          resourceChanged = addComplexAttribute(resource, schemaAttribute, value, extensionUri);
         }
         else if (schemaAttribute.isMultiValued())
         {
-          changeWasMade.weakCompareAndSet(false, addMultivaluedAttribute(resource, schemaAttribute, value));
+          resourceChanged = addMultivaluedAttribute(resource, schemaAttribute, value);
         }
         else
         {
-          changeWasMade.weakCompareAndSet(false, addSimpleAttribute(resource, schemaAttribute, value));
+          resourceChanged = addSimpleAttribute(resource, schemaAttribute, value);
         }
+        verifyReadOnlyNotModified(schemaAttribute, resourceChanged, isReadOnly);
+        changeWasMade.weakCompareAndSet(false, resourceChanged);
       }
     });
     return changeWasMade.get();
   }
 
+  /**
+   * determines if the given json node is read-only or immutable and previously set, if so returns true
+   *
+   * @param jsonNode the node that should be verified
+   * @param schemaAttribute the schema definition of the node
+   */
+  private boolean isReadOnly(JsonNode jsonNode, SchemaAttribute schemaAttribute)
```

Comment:
please rename method to: 

isReadOnlyAndSet

Reply:
👍